src/test/java/com/thirdcc/webapp/web/rest/EventActivityResourceIT.java

```diff
@@ -110,52 +123,205 @@ public static EventActivity createEntity(EntityManager em) {
+        event = createEventEntity(em);
+        eventActivity = createEventActivityEntity(em);
+    }
+
+    @AfterEach
+    public void cleanUp() {
+        eventRepository.deleteAll();
+        eventActivityRepository.deleteAll();
     }
 
     @Test
-    @Transactional
     public void createEventActivity() throws Exception {
+        Event savedEvent = initEventDB();
+
         int databaseSizeBeforeCreate = eventActivityRepository.findAll().size();
 
-        // Create the EventActivity
-        EventActivityDTO eventActivityDTO = eventActivityMapper.toDto(eventActivity);
+        EventActivityDTO eventActivityDTO = createDefaultEventActivityDTO();
+        eventActivityDTO.setEventId(savedEvent.getId());
+
         restEventActivityMockMvc.perform(post("/api/event-activities")
             .contentType(TestUtil.APPLICATION_JSON_UTF8)
             .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
             .andExpect(status().isCreated());
 
-        // Validate the EventActivity in the database
         List<EventActivity> eventActivityList = eventActivityRepository.findAll();
         assertThat(eventActivityList).hasSize(databaseSizeBeforeCreate + 1);
         EventActivity testEventActivity = eventActivityList.get(eventActivityList.size() - 1);
-        assertThat(testEventActivity.getEventId()).isEqualTo(DEFAULT_EVENT_ID);
+        assertThat(testEventActivity.getEventId()).isEqualTo(savedEvent.getId());
         assertThat(testEventActivity.getStartDate()).isEqualTo(DEFAULT_START_DATE);
-        assertThat(testEventActivity.getDurationInDay()).isEqualTo(DEFAULT_DURATION_IN_DAY);
+        assertThat(testEventActivity.getDurationInDay()).isEqualTo(DEFAULT_DURATION_IN_DAY.setScale(2));
         assertThat(testEventActivity.getName()).isEqualTo(DEFAULT_NAME);
         assertThat(testEventActivity.getDescription()).isEqualTo(DEFAULT_DESCRIPTION);
     }
 
     @Test
-    @Transactional
+    public void createEventActivity_WithEventEnded_ShouldThrow400() throws Exception {
+        event.setEndDate(Instant.now().minus(1, ChronoUnit.DAYS));
+        Event savedEvent = initEventDB();
+
+        int databaseSizeBeforeCreate = eventActivityRepository.findAll().size();
+
+        assertThat(savedEvent.getEndDate()).isBefore(Instant.now());
+
+        EventActivityDTO eventActivityDTO = createDefaultEventActivityDTO();
+        eventActivityDTO.setEventId(savedEvent.getId());
+
+        restEventActivityMockMvc.perform(post("/api/event-activities")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeCreate);
+    }
+
+    @Test
+    public void createEventActivity_WithEventClosed_ShouldThrow400() throws Exception {
```

Comment:
@kachun333 
Event is Closed might be meaning that it not allow more participant
- due to limit number of seat maybe
so i think it should be allow to add eventActivity
Or Else we just remove CLOSED for now since we will not using it yet
@LUXIANZE please help clarify on this

Reply:
This is a business decision. 
@LUXIANZE can you please make a decision?

Reply:
@HowardYaw I think, since we do keep a number of participant. we dont actually need this enum to indicate it is closed. Closed is when the event is finished.

Reply:
@LUXIANZE 
if closed mean the event finish:
- then mean we might need a cron jobs to check whether there is event endDate < now()
- if yes then we set the status to CLOSED

I will suggest to just remove CLOSED status
@Five-Fishes/omni-team any thought?

Reply:
Remove CLOSED status
Not using for now
Event endDate is used to check the event is past